Panic on `MessageQueue.Push` in status-go tests

# Problem

Seems that https://github.com/status-im/status-go/pull/5523 introduced some flaky tests 🙁  

For example for test `TestCommunityTokensMetadata`.

```go
panic: send on closed channel

goroutine 228896 [running]:
github.com/waku-org/go-waku/waku/v2/api/publish.(*MessageQueue).Push(...)
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/api/publish/message_queue.go:131
github.com/status-im/status-go/wakuv2.(*Waku).Send(0x14002e9cf00, {0x14000bb8400?, 0xa?}, 0x1400347a000?, 0x107a22d50)
        /Users/igorsirotin/Repositories/Status/status-go/wakuv2/message_publishing.go:56 +0x140
github.com/status-im/status-go/wakuv2.(*PublicWakuAPI).Post(0x14006e4a3f0, {0x14001258f00?, 0x84?}, {{0x14001259100, 0x40}, {0x0, 0x0, 0x0}, {0x14001258f00, 0x40}, ...})
        /Users/igorsirotin/Repositories/Status/status-go/wakuv2/api.go:259 +0x568
g
```

--- a/go.mod
+++ b/go.mod
@@ -96,7 +96,7 @@ require (
 	github.com/schollz/peerdiscovery v1.7.0
 	github.com/siphiuel/lc-proxy-wrapper v0.0.0-20230516150924-246507cee8c7
 	github.com/urfave/cli/v2 v2.27.2
-	github.com/waku-org/go-waku v0.8.1-0.20240806122111-5aa11311f833
+	github.com/waku-org/go-waku v0.8.1-0.20240806200653-4f1d692413e9
 	github.com/wk8/go-ordered-map/v2 v2.1.7
 	github.com/yeqown/go-qrcode/v2 v2.2.1
 	github.com/yeqown/go-qrcode/writer/standard v1.2.1

--- a/wakuv2/message_publishing.go
+++ b/wakuv2/message_publishing.go
@@ -53,9 +53,15 @@ func (w *Waku) Send(pubsubTopic string, msg *pb.WakuMessage, priority *int) ([]b
 	envelope := protocol.NewEnvelope(msg, msg.GetTimestamp(), pubsubTopic)
 
 	if priority != nil {
-		w.sendQueue.Push(envelope, *priority)
+		err := w.sendQueue.Push(w.ctx, envelope, *priority)
+		if err != nil {
+			return nil, err
+		}
 	} else {
-		w.sendQueue.Push(envelope)
+		err := w.sendQueue.Push(w.ctx, envelope)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	w.poolMu.Lock()
@@ -75,7 +81,7 @@ func (w *Waku) broadcast() {
 		var envelope *protocol.Envelope
 
 		select {
-		case envelope = <-w.sendQueue.Pop():
+		case envelope = <-w.sendQueue.Pop(w.ctx):
 
 		case <-w.ctx.Done():
 			return

